src/views/domain-page/domain-page-cluster-selector/domain-page-cluster-selector.tsx

```diff
@@ -0,0 +1,48 @@
+'use client';
+import React from 'react';
+import { Select, SIZE } from 'baseui/select';
+import { Route } from 'next';
+import { useRouter, useParams } from 'next/navigation';
+
+import { overrides, styled } from './domain-page-cluster-selector.styles';
+import { DomainHeaderInfoItemContentProps } from '../domain-page-header-info/domain-page-header-info.types';
+
+export default function DomainPageClusterSelector(
+  props: DomainHeaderInfoItemContentProps
+) {
+  const router = useRouter();
+  const { domain, domainTab } = useParams();
+
+  if (props.domainInfo.clusters?.length === 1) {
+    return <styled.ItemLabel>{props.cluster}</styled.ItemLabel>;
+  }
+
+  return (
+    <Select
+      overrides={overrides.select}
+      options={props.domainInfo.clusters.map((cluster) => ({
+        id: cluster.clusterName,
+        label: cluster.clusterName,
+      }))}
+      value={[
+        {
+          id: props.cluster,
+          label: props.cluster,
+        },
+      ]}
+      onChange={({ option }) => {
+        if (option && option.id !== props.cluster) {
+          const newPath =
+            `/domains/${domain}/${option.id}/${domainTab}` as Route;
```

Comment:
`option.id` needs to be url encoded

Reply:
Route needs to be imported as type. I can see this in multiple places so you just go through them